osmtm/security.py

```diff
@@ -23,32 +23,52 @@ class RootFactory(object):
     __acl__ = [
         (Allow, Everyone, 'view'),
         (Allow, Everyone, 'project_show'),
+        (Allow, Everyone, 'task_contribute'),
         (Allow, 'group:admin', 'add'),
         (Allow, 'group:admin', 'license_edit'),
         (Allow, 'group:admin', 'user_edit'),
         (Allow, 'group:admin', 'project_edit'),
+        (Allow, 'group:admin', 'task_contribute'),
         (Allow, 'group:project_manager', 'project_edit'),
+        (Allow, 'group:project_manager', 'task_contribute'),
     ]
 
     def __init__(self, request):
         if request.matchdict and 'project' in request.matchdict:
             project_id = request.matchdict['project']
             project = DBSession.query(Project).get(project_id)
             if project is not None:
+                if project.private:
```

Comment:
Reviewing the code now, this will require modification. Since `__acl__` uses the permissions defined first-most, and we prepend the acl list as we advance in the code, the private project block should move back to where it was but converted into an `if` instead of retaining the `elif` statement. This `if` distinction is required to prevent users assigned to a project that is now `closed` from accessing it.